src/main/java/com/pubnub/api/managers/RetrofitManager.java

```diff
@@ -124,6 +127,20 @@ public RetrofitManager(PubNub pubNubInstance) {
         this.messageActionService = transactionInstance.create(MessageActionService.class);
         this.filesService = transactionInstance.create(FilesService.class);
         this.s3Service = noSignatureInstance.create(S3Service.class);
+
+        this.pubnub.addListener(new SubscribeCallback.BaseSubscribeCallback() {
+            @Override
+            public void status(@NotNull final PubNub pubnub, @NotNull final PNStatus pnStatus) {
+                if (pnStatus.getCategory() == PNStatusCategory.PNReconnectedCategory) {
+                    Executors.newSingleThreadExecutor().execute(new Runnable() {
```

Comment:
Don't you think it could be better as a field? I mean the `executor` and here only `executor.execute(...)`?